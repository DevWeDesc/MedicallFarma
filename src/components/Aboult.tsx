import React from "react";
import ImagesGrid from "../../public/ImagesGrid.png";
import Image from "next/image";

export const Aboult = () => {
  return (
    <div className="flex flex-col-reverse py-12 px-5 gap-5 lg:pt-0 lg:px-0 lg:grid lg:grid-cols-2 lg:gap-20 lg:py-0">
      <Image alt="" className="lg:min-h-[30rem] lg:w-full" src={ImagesGrid} />
      <div className="flex flex-col gap-3 lg:py-8 lg:pr-16 lg:gap-10">
        <div className="flex flex-col lg:gap-5">
          <h2 className="text-xl font-bold lg:text-5xl">
            Sobre a Medicall Farma
          </h2>
          <div className="text-xs font-light lg:text-lg">
            <p>
              Atuamos há mais de 10 anos no mercado de vendas e distribuição de
              materiais médicos e medicamentos hospitalares em todo o estado de
              São Paulo.
            </p>
            <p>
              Oferecemos a melhor experiência de atendimento aos nossos
              clientes, fornecedores e acionistas, dede o primeiro contato.
              Trabalhamos com um mix de Produtos.
            </p>
            <p>
              Diversificados das melhores marcas nacionais e internacionais
              garantimos total adilidade e pontualidade nas entregas através de
              um rigoroso processo de qualidade operacional.
            </p>
          </div>
        </div>
        <div className="flex flex-col lg:gap-5">
          <h2 className="text-xl font-bold lg:text-5xl">Nossa Missão</h2>
          <p className="text-xs font-light lg:text-lg">
            A Medicall Farma tem como missão contribuir com a melhoria da vida e
            saúde das pessoas, por meio de fornecimento de materiais médicos e
            medicamentos hospitalares de alta qualidade às instituições de saúde
            privadas e públicas do estado de São Paulo.
          </p>
        </div>
      </div>
    </div>
  );
};
