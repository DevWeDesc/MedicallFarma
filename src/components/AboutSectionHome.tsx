import React from "react";
import imageMedicall from "../../public/imagemedicall.jpeg";
import Image from "next/image";
import Link from "next/link";
import { AnimationScroll } from "@/animations/AnimationScroll";

export const AboutSectionHome = () => {
  return (
    <AnimationScroll>
      <div className="flex flex-col justify-center relative items-center bg-black text-white">
        <Image
          alt=""
          className="w-full object-cover h-80 opacity-60"
          src={imageMedicall}
        />
        <div className="absolute h-full w-full flex flex-col justify-center gap-8 items-center">
          <p className="text-2xl font-bold px-5 text-center">
            Conhe√ßa mais sobre a MEDICALL FARMA
          </p>
          <Link
            className="text-center border w-auto px-8 py-2 rounded-lg hover:font-semibold hover:border-[2px] transition-all"
            href="about"
          >
            Saiba Mais!
          </Link>
        </div>
      </div>
    </AnimationScroll>
  );
};
