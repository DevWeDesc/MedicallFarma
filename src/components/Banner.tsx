"use client";
import React from "react";
// import ImageBanner from "../../public/ImageBanner.png";
import ImageBanner from "../../public/GroupProductsBanner.svg";
import Image from "next/image";
import { motion } from "framer-motion";

export const Banner = () => {
  return (
    <div
      // className="bg-grayDark px-5 flex flex-col-reverse gap-8 py-16 xl:grid xl:grid-cols-2 xl:items-center xl:gap-28 xl:pl-28 xl:pt-24 2xl:pl-80"
      className="bg-grayDark px-5 flex flex-col gap-8 py-16 sm:pt-24 xl:items-center xl:gap-12 xl:pt-24"
    >
      <div className="text-white flex flex-col gap-8">
        <h2 className="max-w-[387px] m-auto text-2xl sm:text-3xl xl:max-w-4xl font-light text-center xl:text-5xl">
          Encontre os melhores Produtos Hospitalares com os Melhores Pre√ßos aqui
          na <span className="font-bold">Medicall Farma!</span>
        </h2>
        {/* <Link
          href="#productRecomended"
          className="xl:max-w-md text-center rounded-full py-3 px-10 font-bold text-xl hover:bg-greenDark transition-all bg-greenDefault"
        >
          Procurar Medicamentos
        </Link> */}
      </div>
      <motion.div
        initial={{ scale: 0 }}
        animate={{ scale: 1 }}
        whileHover={{ scale: 1.1 }}
        transition={{ duration: 1.5 }}
      >
        <Image alt="" className="m-auto" src={ImageBanner} />
      </motion.div>
      {/* <Image alt="" src={ImageBanner} /> */}
    </div>
  );
};
