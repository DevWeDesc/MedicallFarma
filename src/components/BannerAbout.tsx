import Image from "next/image";
import React from "react";
import ImageBannerAbout from "../../public/Rectangle.png";
import ImageBannerAboutMobile from "../../public/ImageAboutMobile.png";

export const BannerAbout = () => {
  return (
    <div className="relative flex flex-col justify-center items-center pt-22">
      <Image
        alt=""
        className="hidden lg:block w-full h-auto"
        src={ImageBannerAbout}
      />
      <Image alt="" src={ImageBannerAboutMobile} className="lg:hidden w-full" />
      <div className="absolute text-white flex flex-col items-center w-full gap-8 px-4">
        <h2 className="max-w-[387px] text-center text-2xl font-bold lg:max-w-4xl lg:m-auto xl:text-5xl">
          Missão de Saúde: Fornecendo Excelência em Medicamentos e Materiais
          Médicos
        </h2>
        <p className="text-sm lg:text-base lg:max-w-4xl lg:m-auto text-center font-light">
          Bem-vindo à MEDICALL FARMA, sua parceira dedicada na entrega de
          soluções de saúde excepcionais. Com uma visão centrada no bem-estar,
          estamos comprometidos em fornecer medicamentos e materiais médicos de
          qualidade superior, promovendo a excelência no cuidado médico.
          Descubra mais sobre nossa história, valores e o compromisso que temos
          em impulsionar a saúde e o progresso na comunidade médica.
        </p>
      </div>
    </div>
  );
};
