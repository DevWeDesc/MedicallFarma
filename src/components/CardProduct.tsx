"use client";
import React, { useEffect, useState } from "react";
import { IProductsData } from "../../types/types";
import Image from "next/image";

export const CardProduct = ({
  image,
  name,
  content,
  supplier,
  categories,
}: IProductsData) => {
  const [configCard, setConfigCard] = useState("");
  const [contentProduct, setContentProduct] = useState("");

  useEffect(() => {
    if (location) {
      location.pathname != "/" && setConfigCard("large");
    }
    configCard === "large"
      ? setContentProduct(content)
      : setContentProduct(content?.substring(0, 83));
  }, [configCard]);

  return (
    <div
      className={`flex flex-col text-left ${
        configCard === "large" ? "gap-4 items-center" : "gap-2"
      } gap-2`}
    >
      <Image alt="" src={image} />
      <div className={`${configCard === "large" && "flex flex-col gap-2"}`}>
        <p
          className={`font-medium min-h-[32px] ${
            configCard === "large" ? "text-2xl" : "text-xs "
          }`}
        >
          {name}
        </p>
        <p
          className={`font-light pr-3 ${
            configCard === "large" ? "text-sm opacity-70" : "text-xs"
          }`}
        >
          {contentProduct}
        </p>
        <p
          className={`font-medium  ${
            configCard === "large" ? "text-base" : "text-xs"
          }`}
        >
          {supplier}
        </p>
        {configCard === "large" && (
          <div>
            <button>Solicitar um Or√ßamento</button>
            <p>
              Categorias:{" "}
              {categories?.map((item, index) => (
                <span key={index}>{item}</span>
              ))}
            </p>
          </div>
        )}
      </div>
    </div>
  );
};
