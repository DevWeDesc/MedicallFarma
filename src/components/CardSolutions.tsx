"use client";
import React, { useEffect, useState } from "react";
import { ICardSoluctions } from "../../types/types";
import Image from "next/image";
import image from "next/image";

export const CardSolutions = ({ image, title }: ICardSoluctions) => {
  const [bgColorCard, setBgColorCard] = useState("");
  useEffect(() => {
    if (title === "Medicamentos") {
      setBgColorCard("bg-blueDark");
    } else if (title === "Dietas Enterais") {
      setBgColorCard("bg-yellowDefault");
    } else {
      setBgColorCard("bg-greenDark");
    }
  }, []);

  return (
    <div
      className={`relative px-2 pt-3 pb-12 rounded-lg ${bgColorCard} hover:bg-opacity-90 transition-all`}
    >
      <p className="text-sm font-semibold">{title}</p>
      <button className="text-[10px] bg-transparent border-b pb-[2px] border-white">
        Conhe√ßa agora
      </button>
      <Image alt="" className="absolute right-2 bottom-2" src={image} />
    </div>
  );
};
