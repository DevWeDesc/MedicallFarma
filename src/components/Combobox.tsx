"use client";
import Image from "next/image";
import Link from "next/link";
import React, { useState } from "react";
import { CategoriesData } from "../../data/data";
import { ICategories } from "../@types/types";

export const Combobox = () => {
  const [comboOpen, setComboOpen] = useState(false);
  const [categorySelected, setCategorySelected] = useState<ICategories>();

  return (
    <div className="hidden sm:flex sm:pt-16 xl:flex flex-col pt-20">
      <div className="bg-grayDark text-white flex gap-10 justify-center">
        {CategoriesData.map((item, index) => (
          <button
            key={index}
            onMouseLeave={() => {
              setCategorySelected(item);
              setComboOpen(false);
            }}
            onMouseOver={() => {
              setCategorySelected(item);
              setComboOpen(true);
            }}
            className={`p-8 bg-grayDark hover:font-bold transition-all ${
              categorySelected?.name === item.name && "font-bold"
            }`}
          >
            {item.name}
          </button>
        ))}
      </div>
      <div
        className="relative flex w-full justify-center"
        onMouseLeave={() => {
          setComboOpen(false);
          setCategorySelected(categorySelected);
        }}
        onMouseOver={() => {
          setComboOpen(true);
          setCategorySelected(categorySelected);
        }}
      >
        <div
          className={`${
            comboOpen && categorySelected
              ? "absolute bg-grayLight z-10 py-10 px-6 rounded-md gap-10 grid grid-cols-2 items-stretch text-white transition-all"
              : "absolute h-0 transition-all overflow-hidden"
          }`}
        >
          <div
            className={`${
              categorySelected && categorySelected?.subCategories?.length >= 11
                ? "grid grid-cols-2"
                : "flex"
            } flex flex-col gap-6`}
          >
            {categorySelected?.subCategories.map((item, index) => (
              <Link
                href={`${item.href}`}
                key={index}
                className="flex items-center gap-3 opacity-90 hover:opacity-100 hover:font-semibold transition-all"
              >
                <p>{item.title}</p>
              </Link>
            ))}
          </div>
          {categorySelected && (
            <div className="w-full flex justify-stretch h-full items-stretch rounded-lg">
              <Image
                alt=""
                className="object-cover w-full rounded-lg"
                src={categorySelected?.image}
              />
            </div>
          )}
        </div>
      </div>
    </div>
  );
};
