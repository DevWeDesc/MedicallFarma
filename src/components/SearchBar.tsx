import React, { useEffect, useRef, useState } from "react";
import { IoIosSearch } from "react-icons/io";
import { ProductsRecomendedData } from "../../data/data";
import { IProductsData } from "../../types/types";
import { CardProduct } from "./Cards/CardProduct";
import Link from "next/link";
import Image from "next/image";
import { IoArrowBackOutline } from "react-icons/io5";

type ISearchBar = {
  handleSearchOpen: () => void;
};

export const SearchBar = ({ handleSearchOpen }: ISearchBar) => {
  const [productSearchFiltered, setProductSearchFiltered] = useState<
    IProductsData[]
  >([]);
  const [valueInput, setValueInput] = useState("");

  useEffect(() => {
    let ProductData: IProductsData[] = [];
    valueInput
      ? (ProductData = ProductsRecomendedData.filter((item) =>
          item.name.toLocaleLowerCase().includes(valueInput.toLocaleLowerCase())
        ))
      : (ProductData = ProductsRecomendedData);

    setProductSearchFiltered(ProductData);
  }, [valueInput]);

  return (
    <div className="bg-grayLight py-1 px-2 gap-2 flex justify-center rounded-full w-full relative">
      <div
        className="h-full flex items-center p-1 text-white justify-center"
        onClick={handleSearchOpen}
      >
        <IoArrowBackOutline />
      </div>
      <input
        value={valueInput}
        onChange={(ev) => setValueInput(ev.target.value)}
        onDoubleClick={handleSearchOpen}
        id="serchBar"
        className="w-full text-[10px] bg-transparent outline-none focus:bg-transparent value:bg-transparent text-white placeholder-gray-300"
        type="text"
        placeholder="Procure Produtos!!"
      />
      <label
        htmlFor="serchBar"
        className="p-1 bg-grayDefault text-white rounded-full"
      >
        <IoIosSearch />
      </label>
      <div className="absolute max-h-96 px-4 rounded-lg z-10 flex flex-col gap-10 py-6 overflow-auto bg-grayLight mt-10">
        {productSearchFiltered.map((item) => (
          <Link
            href={`/detailsProduct/${item.id}`}
            key={item.id}
            onClick={handleSearchOpen}
            className="text-white"
          >
            <div className="flex items-center gap-2">
              <Image alt="" height={32} src={item.image} />{" "}
              <div>
                <p className="text-xs">{item.name}</p>
                <span className="text-[10px]">
                  {item.content.substring(0, 23).concat("...")}
                </span>
              </div>
            </div>
          </Link>
        ))}{" "}
        {productSearchFiltered.length <= 0 && (
          <p className="text-white p-4 text-xs text-center">
            Produto Indispon√≠vel!
          </p>
        )}
      </div>
    </div>
  );
};
