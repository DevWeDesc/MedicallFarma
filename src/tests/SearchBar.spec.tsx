/**
 * @jest-environment jsdom
 */
import { SearchBar } from "@/components/SearchBar";
import { render } from "@testing-library/react";

describe("SearchBar", () => {
  it("renders SearchBar unchanged", () => {
    const { container } = render(<SearchBar handleSearchOpen={() => false} />);

    expect(container).toMatchSnapshot();
  });
});
