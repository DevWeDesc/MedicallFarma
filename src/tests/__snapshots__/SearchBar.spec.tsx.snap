// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders SearchBar unchanged 1`] = `
<div>
  <div
    class="bg-grayLight py-1 px-2 gap-2 flex justify-center rounded-full w-full relative lg:w-2/3 lg:m-auto"
  >
    <div
      class="h-full flex items-center p-1 text-white justify-center xl:hidden"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M244 400L100 256l144-144M120 256h292"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="48"
        />
      </svg>
    </div>
    <input
      class="w-full text-[10px] bg-transparent outline-none focus:bg-transparent value:bg-transparent text-white placeholder-gray-300 xl:p-3 xl:text-sm"
      id="serchBar"
      placeholder="Procure Produtos!!"
      type="text"
      value=""
    />
    <label
      class="p-1 xl:p-2 bg-grayDefault text-white rounded-full xl:text-3xl"
      for="serchBar"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
        />
      </svg>
    </label>
  </div>
</div>
`;
